src/main/java/com/github/codeidoscope/Server.java

```diff
@@ -0,0 +1,51 @@
+package com.github.codeidoscope;
+
+public class Server {
+
+    private ServerConnectionWrapper serverConnection;
+
+    public static void main(String[] args) {
+        if (args.length != 2) {
+            System.err.println("Missing port number");
+            System.exit(1);
+        }
+
+        int portNumber = Integer.parseInt(args[1]);
+
+        ServerConnection serverConnection = new ServerConnection();
+        Server server = new Server(serverConnection);
+        server.startServer(portNumber);
+    }
+
+    public Server(ServerConnectionWrapper serverConnection) {
```

Comment:
As these two following functions are public only for testing purpose, I would extract them out of this file.

This way it would result to have your implementation that with this file initialise the server with the _production_ instance in `main()` and your tests using the very same file and injecting mock